### ISSUE TITLE
Keep parentheses as I wrote them

### ISSUE THREAD

**Issue by arlomedia** [Swiftify-Corp/Swiftify#159]
I'd love to see an option, or a default change, to keep parentheses in conditional expressions just as I wrote them. Here's one example:

input: if ( (a == 1) || ((b == 2)&&(c == 3)) ) {}
output: if (a == 1) || ((b == 2) && (c == 3)) {}
demo: http://swiftify.me/rri0od

In this case, the outer parens were removed, and spaces were added. There are many variations of this in my project, and I've had to carefully go through and adjust the formatting back to the way I wrote it, which is most readable to me.

In short, I want Swiftify to convert the language, but not change my formatting or coding style. :-)

**Comment by alex-swiftify** [comment]
@arlomedia Your report is very well understandable and makes sense, although keeping exactly the same parentheses as in the input code is sometimes questionable.

1) As you understand, Swift doesn't require parentheses around if condition, unlike Objective-C.
Converting `if (boolVar)` to `if (boolVar)` may be considered as redundant.
We'll see if we can make this configurable though.

2) Keeping original spacing is very difficult, especially for cases like your sample, where an expression gets converted to a nil comparison.

3) For your demo sample (when an expression inside parentheses gets converted to nil comparison), it's indeed better to keep parentheses. This could be relatively easy to fix.

Will follow up on this later as Swift 5.4 support keeps our team busy at the moment.

**Comment by arlomedia** [comment]
Thanks for considering. I can hardly imagine how much work you've put into this product already!
